### ISSUE TITLE
vagrant up inclusion

### ISSUE THREAD

**Issue by blaggacao** [camptocamp/puppet-openerp#14]
Hello Guys,

I'd like to make a suggestion for even easier Dev-Environment-Setup via Vagrant. I'm quite new to all of this, but I'd like to put a localizacion community in Colombia all on the same page and I think the best way is via a dead simple vagrant box, which installs a fully fledged and working OpenERP-Server as a VM for development in no time. Best would be some kind of chioce for the OpenERP-Version after doing the `vagrand box add` command.

I would like to link two people to this issue which already have done some greate work on this issue on GitHub

https://github.com/arthru/vagrant_openerp
https://github.com/bmart/vagrant-openerp/

I think when camptocamp best practices could be added, we can get a really watertight Vagrant-Best-Practice-Standard.

**Comment by blaggacao** [comment]
@arthru
@bmart

What do you think? Wouldn't it be worthwile to ask Camptocamp for sharing it's best practices in this way, you already worked on?

**Comment by blaggacao** [comment]
Vagrant-Boxes could be postet at https://vagrantcloud.com/OpenERP - maybe this account should one day be handed over to OpenERP S.A.?

I've created an empty project https://github.com/blaggacao/OpenERP-VagrantUp_select_version - and will try to consolidate the work already done by all of you. I'm not a programmer. But I want to get this going ASAP, so **please help me without pardon** ;)

**Comment by bmart** [comment]
I agree a best practices Pre-built box should be available. My project was
made before this whole vagrant cloud thing existed.  It was more a way to
create a generic build recipe from a minimal vanilla Ubuntu box that could
be used for virtualbox or Amazon.  Is there nothing on the vagrant cloud
right now related to OpenERP?

**Comment by blaggacao** [comment]
Hy @bmart 
in my subsequent search I stumbled over loosely to non-associated groups of people and inidividuals who are working on this topic. Nothing with real momentum so far, though..

I tried to invite everyone to the project I created out of not knowing better how to propagate this idea. There should be an easy way to make it a group project in github though? How could I add you for example?

I got a vagrant setup with a docker image running fine this week end. It's far from best practice installation but I was quite happy, because at least it's a running OpenERP, fast and very easy... (takes 2 mins to run)...

I think we likeminded people should use what vagrant is offering and on top of that use a dual approche of two concurring concepts:
- Virtual Environments with Docker-Boxes
- Provisioning on Virtual Machines with Ansible

I'm not sure, if that is wise or not, but to my humble brain this sounds strategically right.

To answer your question, on the vagrant cloud, the only one is me that created a user named OpenERP. If I've understood it right however, the vagrantcloud is for boxes. The disadvantage - still if I got it right - is that you cannot that easily adapt for other versions of programs or adapt the config, which is already the next step in the learning courve after installing a vanilla setup.

I think for this purpose anyhow, Docker-Images is much better suited, as it can be run in a less ressource intensive Virtual Environment (or in a VM optionally).

If I might dream, this would be a modular ansible blaybook, which can handle "vanilla-dev", "vanilla-dev with batteries", "high security producion" provisioning, all using communities best practices complemented by one or more instand-ready Docker-Images which are mutually synced...
Dockers would be fine for the majority of OpenERPlers and Ansibles could be used and adapted for hosting providers and more advanced players...

Maybe we should launch into some consolidation work over here:
https://github.com/blaggacao/OpenERP-VagrantUp

@camptocamp hope the swiss masters will join :+1:
